app/concepts/lockstep/client.rb

```diff
@@ -29,6 +29,13 @@ def self.api_key
       RequestStore.store[:lockstep_api_key]
     end
 
+    def self.set_magic_link_secret(key)
+      RequestStore.store[:magic_link_secret] = key
```

Comment:
I think this secret might be used by all internal API calls, beyond magic link. Could you rename this to something more generic?

Reply:
ex: `internal_service_key` since the header is internal service